Domain.FSharp/Shipping/Location.fs

```diff
@@ -0,0 +1,24 @@
+namespace FSharp.Domain.Shipping.Location
+
+open System
+open System.Text.RegularExpressions
+
+type UnLocode(value: string) =
+
+    do
+        if isNull value then raise (ArgumentNullException("value"))
+
+    do
+        let pattern = Regex "[a-zA-Z]{2}[a-zA-Z2-9]{3}"
+        let matches = pattern.Match value
+        if not matches.Success then raise (ArgumentException("Provided value is not a valid UnLocode", "value"))
+
+    member val Value = value
+
+type Location(unlocode: UnLocode, name: string) =
+
+    do
+        if String.IsNullOrWhiteSpace(name) then raise (ArgumentException("Provided name is not valid", "name"))
```

Comment:
Also, on `IsNullOrWhiteSpace`, see https://blog.ploeh.dk/2014/11/18/the-isnullorwhitespace-trap/.

Reply:
I read the article and I agree. Let's address that when we will introduce more functional concepts.